src/DripsHub.sol

```diff
@@ -453,4 +453,11 @@ contract DripsHub is Managed {
             storageRef.slot := slot
         }
     }
+
+    /// @notice Generates an asset ID for the ERC-20 token
+    /// @param erc20 The ERC-20 token
+    /// @return assetId The asset ID
+    function _assetId(IERC20 erc20) internal pure returns (uint256 assetId) {
+        return uint160(address(erc20));
+    }
```

Comment:
I still think this might be a bit over-engineered or I forgot why we are doing it. 

Why are we not just using an `address` as asset type?

Reply:
We're doing it to make the Drips library as generic and unconstrained as possible. The library doesn't care what the assetId represents, we're using it for ERC-20 addresses, but others could use it to encode any kinds of things, e.g. they may want to encode drips in Ether, ERC-1155s or even some custom asset IDs. From the performance perspective it doesn't matter as long as it fits in a single word.